java/src/jmri/jmrit/operations/rollingstock/RollingStock.java

```diff
@@ -813,7 +815,7 @@ public void propertyChange(java.beans.PropertyChangeEvent e) {
                         if (e.getNewValue() != null) {
                             Location newLocation = locationManager.getLocationByReporter((jmri.Reporter) e.getNewValue());
                             if (newLocation != getLocation())
-                                setLocation(newLocation, null);
+                                setWhereLastSeen(newLocation);
```

Comment:
Seems to me that two concepts are being conflated, and things will be simpler if they're separated out to different, separate values:

1) last place positively identified, e.g. by an RFID reader

2) last place estimated to be by some tracking process